app/src/androidTest/java/org/mozilla/focus/activity/TakeScreenshotTest.java

```diff
@@ -0,0 +1,129 @@
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.pressImeActionButton;
+import static android.support.test.espresso.action.ViewActions.replaceText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.core.AllOf.allOf;
+
+@Keep
+@RunWith(AndroidJUnit4.class)
+public class TakeScreenshotTest {
+
+    private static final int TIMEOUT_WAIT_ANIMATION_FINISHED = 1000;
+    @Rule
+    public final ActivityTestRule<MainActivity> activityTestRule = new ActivityTestRule<>(MainActivity.class, true, false);
+
+    @Rule
+    public final GrantPermissionRule writePermissionRule = GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE);
+
+    @Rule
+    public final GrantPermissionRule readPermissionRule = GrantPermissionRule.grant(Manifest.permission.READ_EXTERNAL_STORAGE);
+
+    private static final String TARGET_URL_SITE = "file:///android_asset/gpl.html";
+
+    @Before
+    public void setUp() {
+        AndroidTestUtils.beforeTest();
+    }
+
+    @After
+    public void tearDown() {
+        activityTestRule.getActivity().finishAndRemoveTask();
+    }
+
+    @Test
+    public void takeScreenshot_screenshotIsCaptured () {
+
+        activityTestRule.launchActivity(new Intent());
+
+        final SessionLoadedIdlingResource loadingIdlingResource = new SessionLoadedIdlingResource(activityTestRule.getActivity());
+
+        // Click search field
+        onView(allOf(withId(R.id.home_fragment_fake_input), isDisplayed())).perform(click());
+
+        // Enter test site url
+        onView(allOf(withId(R.id.url_edit), isDisplayed())).perform(replaceText(TARGET_URL_SITE), pressImeActionButton());
+
+        // Check if test site is loaded
+        IdlingRegistry.getInstance().register(loadingIdlingResource);
+        onView(allOf(withId(R.id.display_url), isDisplayed())).check(matches(withText(TARGET_URL_SITE)));
+        IdlingRegistry.getInstance().unregister(loadingIdlingResource);
+
+        final ScreenshotIdlingResource screenshotIdlingResource = new ScreenshotIdlingResource(activityTestRule.getActivity());
+        screenshotIdlingResource.registerScreenshotObserver();
+
+        // Click screen capture button
+        onView(allOf(withId(R.id.btn_capture), isDisplayed())).perform(click());
+
+        // Register screenshot taken idling resource and wait capture complete
+        IdlingRegistry.getInstance().register(screenshotIdlingResource);
+
+        IdlingRegistry.getInstance().unregister(screenshotIdlingResource);
+
+        // Wait a while to get animation dimiss
+        SystemClock.sleep(TIMEOUT_WAIT_ANIMATION_FINISHED);
```

Comment:
We need to disable animation in Android system setting (developer option) first.
This method is not ideal but there is some issue with our CI server. so I hesitate to approve it.
https://developer.android.com/training/testing/espresso/setup.html